CDLUC3/dmsp_aws_prototype
# fix for related works issues in Uc3DmpId::Updater and the DataCiteHar…

Fix for DMPTool issue #626

**Changes:**
- Updated the DataCiteHarvester lambda function so that it includes the `domain` for the related work. This is used by the `Uc3DmpId::Updater` class in the `PutDmp` API call to match approvals/rejections made in the DMPTool to harvested works in the Dynamo table
- Updated the `Uc3DmpId::Updater` class to work with the new `domain` added by the DataCiteHarvester.
- Added a script to delete all existing `HARVESTER_MODS` entries and trigger a new harvest to rebuild them

**Deployment**
- Deploy the changes to the Harvester: `cd src/lambdas/harvesters/datacite && ruby sam_build_deploy.rb [env] true true [log_level]`
- Need to rebuild all existing `HARVESTER_MODS` records in the Dynamo table so that the discovered related works include a `domain`. To do this run `cd src/scripts && ruby reharvest-related-works.rb stg [TableName] [EventBusARN]`
- Deploy the `Uc3DmpId::Updater` gem and update all of the Lambdas: `cd src/sam && ruby sam_build_deploy.rb [env] true true [log_level]`

**Testing**
- Login to the DMPTool and go to the DMP Upload page
- Select a DMP that has related works
- Go to the Related Works section
- Make changes to at least one related work (approve or reject it)
- Inspect the Network activity 
- Click 'Save'
- Make sure you receive a 200 response and that the page redirects to the plan overview 
- Open the AWS Console. and go to DynamoDB
- Select the apporpiate table for the env
- Search for the PK `DMP#doi.org/[DOI]`
- Examine the record with the `SK = 'HARVESTER_MODS'`
- Make sure the status of the related work(s) you changed is correct
- Examine the record with the `SK = 'VERSION#latest'`
- Make sure it's `dmproadmap_related_identifiers` contains the related work (if you approved it) and does not if you 'rejected' it or left it as 'pending'
- Make sure the `dmphub_modification` is not in this record.
